train/DirectoryStatistics.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace DirectoryStatistics
+{
+    class day5
+    {
+        public static long length;
+        public static double size;
+        //save the length and size of the directory
+        public static void Main(string[] args)
+        {
+            /*
+            #if DEBUG
+                        args = new[] { "-d /Users/admin/Desktop/dailyreport -p *.txt" };
+            #endif
+            */
+            string filePath;
+            string pattern;
+            Stopwatch watch = new Stopwatch();
+            watch.Start();
+            if (args.Length > 0 && args[0] == "-d" && args[2] == "-p")
+            {
+                filePath = args[1];
+                pattern = args[3];
+                Console.WriteLine("File name pattern: {0}", pattern);
+                /*for (int i = 0; i < args.Length; i++) {
+                    Console.WriteLine(args[i]);
+                }*/
+                //debug
+                string[] files = Directory.GetFiles(filePath, pattern, SearchOption.AllDirectories);
+                if (files.Length == 0)
+                {
+                    throw new Exception("No suitable file");
```

Comment:
I think if there is no required file under the directory, a better way is to still print normal statistics messages instead of regarding it as an exception:
File name pattern: *.txt
File count: 0
Total size: 0 MB (0 bytes)
Time elapsed: xx ms
File list:
<Empty>